Jenkinsfile

```diff
@@ -0,0 +1,183 @@
+
+      version = readVersion()
+
+      def apiToken
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+        apiToken = env.GITHUB_API_PASSWORD
+      }
+      gitHub = new GitHub(this, "${organization}/${gitHubRepository}", apiToken)
+
+      if (params.nexus_repository_manager_version) {
+        stage('Update Repository Manager Version') {
+          OsTools.runSafe(this, "git checkout ${branch}")
+          dockerImages.each { updateRepositoryManagerVersion(it.dockerFilePath) }
+          version = params.nexus_repository_manager_version
+        }
+      }
+    }
+    stage('Build') {
+      gitHub.statusUpdate commitId, 'pending', 'build', 'Build is running'
+      dockerImages.each {
+        def hash = OsTools.runSafe(this, "docker build --quiet --no-cache --tag ${it.imageTag} -f ${it.dockerFilePath} .")
+        it.imageTag = hash.split(':')[1]
+
+        if (currentBuild.result == 'FAILURE') {
+            gitHub.statusUpdate commitId, 'failure', 'build', 'Build failed'
+            return
+        } else {
+            gitHub.statusUpdate commitId, 'success', 'build', 'Build succeeded'
+        }
+      }
+    }
+    if (params.nexus_repository_manager_version) {
+      stage('Commit Repository Manager Version Update') {
+        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'integrations-github-api',
+                        usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+          def commitMessage = params.nexus_repository_manager_version ? "Update Repository Manager to ${params.nexus_repository_manager_version}." : ""
+          OsTools.runSafe(this, """
+            git add .
+            git commit -m '${commitMessage}'
+            git push https://${env.GITHUB_API_USERNAME}:${env.GITHUB_API_PASSWORD}@github.com/${organization}/${gitHubRepository}.git ${branch}
+          """)
+        }
+      }
+    }
+    stage('Archive') {
+      dir('build/target') {
+        dockerImages.each {
+            OsTools.runSafe(this, "docker save ${it.imageTag} | gzip > ${archiveName}-${it.flavor}.tar.gz")
+        }
+        archiveArtifacts artifacts: "${archiveName}-*.tar.gz", onlyIfSuccessful: true
+      }
+    }
+    if (branch != 'master') {
+      return
+    }
+    input 'Push image and tags?'
+    stage('Push image and tags') {
+      def dockerhubApiToken
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'docker-hub-credentials',
+            usernameVariable: 'DOCKERHUB_API_USERNAME', passwordVariable: 'DOCKERHUB_API_PASSWORD']]) {
+        dockerImages.each {
+            imageId = it.imageTag
+            def tags = getTags(it.flavor, version)
+            tags.each {
+                OsTools.runSafe(this, "docker tag ${imageId} ${organization}/${dockerHubRepository}:${it}")
```

Comment:
Since the definition of `it` changes (given closure in closure), I prefer a more explicit variable name for the context.
```
dockerImages.each { image ->
    tags.each { tag ->
    }
}
```

Reply:
In: d83a8b5